src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -1098,3 +1124,60 @@ func isRedpandaClusterVersionManaged(
 	}
 	return true
 }
+
+func patchStatus(ctx context.Context, c client.Client, observedCluster *vectorizedv1alpha1.Cluster, mutator func(cluster *vectorizedv1alpha1.Cluster)) (vectorizedv1alpha1.ClusterStatus, error) {
+	name := types.NamespacedName{
+		Namespace: observedCluster.Namespace,
+		Name:      observedCluster.Name,
+	}
+	cluster := &vectorizedv1alpha1.Cluster{}
+	err := c.Get(
```

Comment:
For performing a patch request, I don't think you'd need to bother refreshing the cluster here. You just trying to push the changes performed by `mutator` up to the server and `Patch` doesn't return errors on ResourceVersion mismatches.